### PR TITLE
Update invocation and execution diagrams

### PR DESCRIPTION
**Pull request by mbrandenburger**
**What this PR does / why we need it**:

* Revist based on HLGF meeting notes
* Cleanup and printer friendly edits
* Remove cc2cc calls from execution diagram
* Resolve open questions regarding response encryption

**Which issue(s) this PR fixes**:
This PR addresses  #355 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing changes and/or breaks backward compatability?**:

```

### CODE REVIEW

File: docs/design/fabric-v2+/fpc-cc-invocation.puml [hunk 1]
```diff
@@ -35,74 +35,103 @@ participant "ERCC Peer1_Org1" as ERCC1 order 50
...
       return
-    else
-      |||
+      return
+  end
 
-      ECC1 -> Enclave1 ++: invoke (proposal(args_ECC))
+      Peer -> ECC1 ++: invoke (proposal(args_ECC), CTX)
+      note right
+        Establish context (CTX) used for all interactions
```
Comment (g2flyer) [inline docs/design/fabric-v2+/fpc-cc-invocation.puml:115]: Maybe calling this "transaction context" with acronym "TCtx" or "TX-CTX" or alike would make the purpose of this more explicit

File: docs/design/fabric-v2+/fpc-cc-invocation.puml [hunk 2]
```diff
@@ -35,74 +35,103 @@ participant "ERCC Peer1_Org1" as ERCC1 order 50
...
+
+CLI -> ERCC1 ++: query getECCInfo(CC_NAME)
+return ECCInfo = List<CC_S_PK, CC_E_PK, ... , ATTESTATION>, ...>
+
+CLI -> CLI: verify ECCInfo
+note right
+  NOTE ECCInfo contains a list of all FPC enclave registered for CC_NAME
+
+  Verification is only necessary iff User is not member of Org1,
+  otherwise authenticated response enough. In that case, the user
```
Comment (g2flyer) [inline docs/design/fabric-v2+/fpc-cc-invocation.puml:54]: What is actually not necessarily obvious what `verify` means, maybe mention that this means doing the attestation verification and alike (essentially what ercc does, so we could explicitly reference the part in ercc which does it and say that the client would have to replicate these steps (with the ground truth of IAS keys and tlcc of course based on some local state)

File: docs/design/fabric-v2+/fpc-cc-invocation.puml [hunk 3]
```diff
@@ -35,74 +35,103 @@ participant "ERCC Peer1_Org1" as ERCC1 order 50
...
 
-  User1 -> User1 : args_ECC1 <- encrypt args with ECC1_E_PK
+group invoke
+  CLI -> Peer ++: send proposal <CC_NAME, args_ECC1,....>
 
-  User1 -> ECC1 ++: send proposal (args_ECC1)
+  alt FPC Chaincode is not running
+      note right Peer
+          handled implicitly by the peer which determines
+          when to call ‘run’ from external builder.
```
[the remark this answers is not in the document]
Comment (mbrandenburger) [reply docs/design/fabric-v2+/fpc-cc-invocation.puml:78]: oh wired. Ok removed

File: docs/design/fabric-v2+/fpc-cc-invocation.puml [hunk 4]
```diff
@@ -111,62 +140,60 @@ loop for all/"enough" endorsers
...
-create entity Orderer order 100
-User1 -> Orderer : send transaction
+return result
 
-deactivate User1
 |||
 
 ref over User1, Orderer
     continue with validation phase after ordering
     see fpc-validation-msp.puml
```
[the remark this answers is not in the document]
Comment (mbrandenburger) [reply docs/design/fabric-v2+/fpc-cc-invocation.puml:195]: ok, just removed the link to msp validation

File: docs/design/fabric-v2+/fpc-cc-invocation.puml [hunk 5]
```diff
@@ -35,74 +35,103 @@ participant "ERCC Peer1_Org1" as ERCC1 order 50
...
+      note right Enclave1
+          protocol is SIGMA with 1-way local attestation of TLCC
+          with reportbody = <CH_id, TLCC_inst_id> where
+          TLCC_inst_id as unique (randomly generated) identifier of
+          this particular TLCC __instance__.
+
+          Verifies local attestation against TLCC_MR and CH_id
+          from sealed state. Remembers TLCC_inst_id.
       end note
       return
```
[the remark this answers is not in the document]
Comment (mbrandenburger) [reply docs/design/fabric-v2+/fpc-cc-invocation.puml:109]: Ah right, ok I move channel creation inside invoke

File: docs/design/fabric-v2+/fpc-cc-invocation.puml (hunk 1 above)
Comment (mbrandenburger) [reply docs/design/fabric-v2+/fpc-cc-invocation.puml:115]: > Maybe calling this "transaction context" with acronym "TCtx" or "TX-CTX" or alike would make the purpose of this more explicit

Transaction context makes sense and I updated the note! I would like to keep ctx as it is nice and short.

File: docs/design/fabric-v2+/fpc-cc-invocation.puml (hunk 2 above)
Comment (mbrandenburger) [reply docs/design/fabric-v2+/fpc-cc-invocation.puml:54]: ok added some clarification and liked to fpc-registration.puml

**Comment by mbrandenburger** [comment]
Updated based on feedback from @g2flyer and recent discussion regarding component naming convention as discussed in https://github.com/hyperledger-labs/fabric-private-chaincode/pull/375_.

@bvavala Please consider for merging.

File: docs/design/fabric-v2+/fpc-cc-invocation.puml (hunk 4 above)
[the remark this answers is not in the document]
Comment (mbrandenburger) [reply docs/design/fabric-v2+/fpc-cc-invocation.puml]: added link to validation diagram and pushed the corresponding diagram here. See PR #378

File: docs/design/fabric-v2+/fpc-cc-invocation.puml (hunk 5 above)
[the remark this answers is not in the document]
Comment (g2flyer) [reply docs/design/fabric-v2+/fpc-cc-invocation.puml]: oops, i guess too late: i should have said more explicitly that one channel per lifetime is ok, too (as you have added the CTX) and it actually made the diagrams actually quite nice but reducing the common path for invoke and adds some "action" to the case `run` is called ...

[the remark this answers is not in the document]
Comment (g2flyer) [reply docs/design/fabric-v2+/fpc-cc-invocation.puml]: looking now at the current version, it looks good too :-)

**Comment by g2flyer** [comment]
> Updated based on feedback from @g2flyer and recent discussion regarding component naming convention as discussed in #375_.
> 
> @bvavala Please consider for merging.

You have a fixup commit! But Bruno obviously can also squash-merge (just needs some manual commit info cleanup but here it should be straight-forward).

File: docs/design/fabric-v2+/fpc-cc-invocation.puml [hunk 6]
```diff
@@ -17,156 +17,173 @@ title Foot Box removed
...
-  In this case, though, assuming client has a reference ECC_MR, TLCC_MR, CH_id
-  it might be sufficient to perform verification done in ""verify and store"" by peers in
-  ""register attestation"" flow.
-  (TBD)
+User1 -> CLI ++: invoke <CC_NAME,..., args>
+
+group prepare FPC proposal
+
+CLI -> ERCC1 ++: query getECCInfo(CC_NAME)
+return ECCInfo = List<CC_S_PK, CC_E_PK, ... , ATTESTATION>, ...>
```
[the remark this answers is not in the document]
Comment (mbrandenburger) [reply docs/design/fabric-v2+/fpc-cc-invocation.puml]: updated! should be in sync with key-dist diagram

File: docs/design/fabric-v2+/fpc-cc-invocation.puml [hunk 7]
```diff
@@ -17,156 +17,179 @@ title Foot Box removed
...
       return
 
 
-      Enclave1 -> Enclave1 : args <- decrypt args_ECC
+      Enclave1 -> Enclave1 : <args, r_key> <- decrypt args_enc with Chaincode_DK
 
       note right
-        If proposal contains optional Req_TLCC_inst_id parameter,
-        make sure it matches own TLCC_inst_id, otherwise abort.
+        If proposal contains optional Req_ledger_enclave_inst_id parameter,
```
Comment (bvavala) [inline docs/design/fabric-v2+/fpc-cc-invocation.puml:126]: This is a good point to address, though it's questionable
1) whether different chaincodes should ever access different TLCC
2) whether the client should define what's appropriate.
At a first glance, my preference would be to implement
* credential-release policies for ERCC, e.g., ERCC gives chaincode credentials to the client only if the chaincode's TLCC mrenclave matches the current one in the registry.
* make the chaincode essentially re-register its updated definition

File: docs/design/fabric-v2+/fpc-cc-invocation.puml [hunk 8]
```diff
@@ -17,156 +17,179 @@ title Foot Box removed
...
+  NOTE ECCInfos contains a list of all FPC enclave registered for CC_Id.
 
-  User1 -> ECC1 ++: send proposal (args_ECC1)
+  Verify performs the same verification steps (i.e., attestation
+  verification, etc.) as the registration step performed by ERCC.
+  See fpc-registration.puml for more details.
 
-  group invoke
+  Note that this verification is only necessary iff User is not member
+  of Org1, otherwise authenticated response enough. In that case, the
```
[the remark this answers is not in the document]
Comment (g2flyer) [reply docs/design/fabric-v2+/fpc-cc-invocation.puml:51]: > * does the client trust the fabric network?
> * how can the client establish trust in TLCC and the currently committed ledger data?

A client of an org interacts (authentically/securely) with peers of its own org. So he can trust that (the standard fabric code!) peer has properly validated all transactions

File: docs/design/fabric-v2+/fpc-cc-invocation.puml (hunk 7 above)
Comment (g2flyer) [reply docs/design/fabric-v2+/fpc-cc-invocation.puml:126]: > * whether different chaincodes should ever access different TLCC

The reason for this field afair is more to ensure they use the _same_ tlcc (mrenclave &) instance (the dishonest peer could try to have different tlcc instances with different views for a single cc2cc tx)

**Pull request merged**